src/main/webapp/app/overview/course-registration/course-registration.component.ts

```diff
@@ -2,35 +2,76 @@ import { Component, OnInit } from '@angular/core';
 import { AccountService } from 'app/core/auth/account.service';
 import { Course } from 'app/entities/course.model';
 import { CourseManagementService } from 'app/course/manage/course-management.service';
-import { faCheckCircle } from '@fortawesome/free-solid-svg-icons';
+import { faCheckCircle, faSort } from '@fortawesome/free-solid-svg-icons';
+import { ASC, DESC, SORT } from 'app/shared/constants/pagination.constants';
+import { ActivatedRoute, Router } from '@angular/router';
+import { UserFilter } from 'app/admin/user-management/user-management.component';
+import { Subject, combineLatest } from 'rxjs';
 
 @Component({
     selector: 'jhi-course-registration-selector',
     templateUrl: './course-registration.component.html',
 })
 export class CourseRegistrationComponent implements OnInit {
+    search = new Subject<void>();
     coursesToSelect: Course[] = [];
     loading = false;
-
-    // Icons
     faCheckCircle = faCheckCircle;
+    faSort = faSort;
+    predicate!: string;
+    ascending!: boolean;
+    searchTermString = '';
+    filters: UserFilter = new UserFilter();
```

Comment:
Is this needed? I again could not find any usages of it and it seems to come from user management, so I am not sure why you would need it here.

Reply:
Only filters is not needed. The others are used by the jhiSort function in the HTML file.